### PR TITLE
[harness-automation] add rf shield device support

### PR DESCRIPTION
**Pull request by maciejnycz**
I added general support for 'RF shield' devices needed during some of the tests. I also added a controller class for RF switch which we use in our test setups. You can add your own rf shield controller classes inheriting from the abstract class RfShieldController.

### CODE REVIEW

File: tools/harness-automation/autothreadharness/rf_shield_controller.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python
+#
+# Copyright (c) 2016, The OpenThread Authors.
```
Comment (bukepo) [inline tools/harness-automation/autothreadharness/rf_shield_controller.py]: `Copyright (c) 2016` -> `Copyright (c) 2017`

File: tools/harness-automation/autothreadharness/harness_case.py [hunk 2]
```diff
@@ -791,14 +806,22 @@ def _handle_dialog(self, dialog, title):
             inp.send_keys(ml64)
 
         elif title.startswith('Shield Devices') or title.startswith('Sheild DUT'):
-            if self.dut and settings.SHIELD_SIMULATION:
+            if self.rf_shield:
+                logger.info("Shielding devices")
```
Comment (bukepo) [inline tools/harness-automation/autothreadharness/harness_case.py]: be consistent with single quote.

File: tools/harness-automation/autothreadharness/rf_shield_controller.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+class RfSwitchController(RfShieldController):
+
+    def __init__(self, channel, port):
+        self._channel = channel
+        self._port = port
+        self._conn = None
+
+    def shield(self):
+        self._display_string("CLOSE {}".format(self._channel))
```
Comment (bukepo) [inline tools/harness-automation/autothreadharness/rf_shield_controller.py]: be consistent with single quote.

File: tools/harness-automation/autothreadharness/rf_shield_controller.py (hunk 1 above)
Comment (maciejnycz) [reply tools/harness-automation/autothreadharness/rf_shield_controller.py]: fixed

File: tools/harness-automation/autothreadharness/rf_shield_controller.py (hunk 3 above)
Comment (maciejnycz) [reply tools/harness-automation/autothreadharness/rf_shield_controller.py]: fixed

File: tools/harness-automation/autothreadharness/harness_case.py (hunk 2 above)
Comment (maciejnycz) [reply tools/harness-automation/autothreadharness/harness_case.py]: fixed

**Pull request merged**